src/Workspaces/MSBuild/BuildHost/BuildHost.cs

```diff
@@ -49,7 +49,11 @@ private bool TryEnsureMSBuildLoaded(string projectOrSolutionFilePath)
                 return true;
             }
 
-            var instance = FindMSBuild(projectOrSolutionFilePath, includeUnloadableInstances: false);
+            // When running in SDK test environment the runtime version might be lower than the SDK version.
+            // We need to force the load of "incompatible" version of the SDK in this case.
```

Comment:
How is MSBuild generally handling this situation? I don't like the idea we have a specific environment variable for this scenario -- we should just do the right thing automatically.

Reply:
They don't handle it because I guess no one needs it except for dotnet-watch. It's a test-only issue in SDK repo.

Reply:
"They don't support this scenario but we want to run tests anyways" doesn't make much sense to me. This is a very temporary window, and having a one off feature to support this seems a bit odd to me. Does this also become moot once we move you to in-proc builds?

My concern being five years from now when we've both won the lottery -- somebody is wondering what this random flag is for and whether it's safe to remove, etc.